ddtrace/writer.py

```diff
@@ -83,21 +83,23 @@ def flush_queue(self):
             traces = self._apply_filters(traces)
         except Exception as err:
             log.error('error while filtering traces: {0}'.format(err))
+            return
 
-        traces_response = None
+        traces_responses = None
 
         if traces:
             # If we have data, let's try to send it.
-            try:
-                traces_response = self.api.send_traces(traces)
-            except Exception as err:
-                log.error('cannot send spans to {1}:{2}: {0}'.format(
-                    err, self.api.hostname, self.api.port))
-
-        if self._priority_sampler and traces_response:
-            result_traces_json = traces_response.get_json()
-            if result_traces_json and 'rate_by_service' in result_traces_json:
-                self._priority_sampler.set_sample_rate_by_service(result_traces_json['rate_by_service'])
+            traces_responses = self.api.send_traces(traces)
+            for response in traces_responses:
+                if isinstance(response, Exception):
+                    log.error('cannot send spans to {1}:{2}: {0}'.format(
```

Comment:
```suggestion
                    log.error('failed to send traces to {1}:{2}: {0}'.format(
```

This isn't really a required change, feel free to ignore, just feels like a better log message

Reply:
Fixed.